src/test/java/com/commercetools/sync/taxcategories/TaxCategorySyncTest.java

```diff
@@ -394,4 +381,87 @@ void sync_WithDuplicatedCountryCode_ShouldNotBuildActionAndTriggerErrorCallback(
+        Assertions.assertThat(errorCallBackMessages).hasSize(1);
+        Assertions.assertThat(errorCallBackExceptions).hasSize(1);
+
+        Assertions.assertThat(errorCallBackMessages.get(0)).contains(
+            format("Failed to update tax category with key: '%s'. Reason: Not found when attempting to fetch while"
+                + " retrying after concurrency modification.", taxCategoryDraft.getKey()));
+    }
+
+    @Test
+    void sync_WithConcurrentModificationExceptionAndFailedFetch_ShouldFailToReFetchAndUpdate() {
+        // preparation
+        List<String> errorCallBackMessages = new ArrayList<>();
+        List<Throwable> errorCallBackExceptions = new ArrayList<>();
+        final TaxCategorySyncOptions spyOptions = TaxCategorySyncOptionsBuilder
+            .of(mock(SphereClient.class))
+            .errorCallback((exception, oldResource, newResource, updateActions) -> {
+                errorCallBackMessages.add(exception.getMessage());
+                errorCallBackExceptions.add(exception.getCause());
+            })
+            .build();
+        final TaxCategoryDraft taxCategoryDraft = TaxCategoryDraftBuilder.of("someName", emptyList(), "changed")
+                                                                         .key("someKey").build();
+        buildTaxCategoryWithConcurrentModificationUpdate();
+        when(taxCategoryService.fetchTaxCategory(any())).thenReturn(supplyAsync(() -> {
+            throw new BadGatewayException();
+        }));
+
+        final TaxCategorySync taxCategorySync = new TaxCategorySync(spyOptions, taxCategoryService);
+
+        // test
+        final TaxCategorySyncStatistics taxCategorySyncStatistics = taxCategorySync
+            .sync(singletonList(taxCategoryDraft))
+            .toCompletableFuture()
+            .join();
+
+        // assertion
+        AssertionsForStatistics.assertThat(taxCategorySyncStatistics).hasValues(1, 0, 0, 1);
+        Assertions.assertThat(errorCallBackMessages).hasSize(1);
+        Assertions.assertThat(errorCallBackExceptions).hasSize(1);
+
+        Assertions.assertThat(errorCallBackExceptions.get(0).getCause()).isExactlyInstanceOf(BadGatewayException.class);
+        Assertions.assertThat(errorCallBackMessages.get(0)).contains(
+            format("Failed to update tax category with key: '%s'. Reason: Failed to fetch from CTP while retrying "
+                + "after concurrency modification.", taxCategoryDraft.getKey()));
+    }
+
+    private void buildTaxCategoryWithConcurrentModificationUpdate() {
+        final TaxCategory taxCategory = mock(TaxCategory.class);
+
+        when(taxCategory.getKey()).thenReturn("someKey");
+        when(taxCategoryService.fetchMatchingTaxCategoriesByKeys(any()))
+            .thenReturn(completedFuture(new HashSet<>(singletonList(taxCategory))));
+        when(taxCategoryService.updateTaxCategory(any(), any())).thenReturn(supplyAsync(() -> {
+            throw new ConcurrentModificationException();
```

Comment:
Have to add following in case update success in retry
```suggestion
        when(taxCategoryService.updateTaxCategory(any(), any()))
        .thenReturn(supplyAsync(() -> {throw new ConcurrentModificationException()}))
        .thenReturn(completedFuture(mockProduct));
```

Reply:
https://github.com/commercetools/commercetools-sync-java/commit/30bc2438b9e4c3d18e635410535472734c0b862d